### DISCUSSION TITLE
How to get ASN ID for vendor shipement to generate shipping label with ASN ID

### DISCUSSION THREAD

**Discussion by zaineco** [amzn/selling-partner-api-models#4168] [Q&A]
I am using POST /vendor/shipping/v1/shipmentConfirmations to submit vendor shipment confirmations. submission is successful but I cannot get the ASN ID of the generated confirmation. 
I need ASN ID to create the Shipping label which contains ASN barcode and shipTo details

any idea would be helpful..

**Comment by jawheral** [comment]
Hi @zaineco,

The Bill of Lading (BOL) is used for appointment booking and shipment label creation. You can see this [here](https://developer-docs.amazon.com/sp-api/docs/vendor-retail-procurement-apis-v1-use-case-guide#country-specific-business-requirements-2).
Please check our [use case guide](https://developer-docs.amazon.com/sp-api/docs/vendor-retail-procurement-apis-v1-use-case-guide) on Vendor Retail Procurement for further details.

I hope this helps!

Jawher
Selling Partner API
